CanHazFunny/CanHazFunny/Jester.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace CanHazFunny;
+public class Jester
+{
+    private Comedian? _Comedian;
+    private JokeService? _JokeService;
+    private string? _Joke;
+    public Comedian? Comedian 
+    {
+        get => _Comedian!;
+        set 
+        { 
+            if (value is null) throw new ArgumentNullException(nameof(value));
+            _Comedian = value;
+        } 
+    }
+    public JokeService? JokeService
+    {
+        get => _JokeService!;
+        set
+        {
+            if (value is null) throw new ArgumentNullException(nameof(value));
+            _JokeService = value;
+        }
+    }
+
+    public string?Joke
+    {
+        get => _Joke!;
+        private set 
+        {
+            if (value is null) throw new ArgumentNullException(nameof(value));
+            _Joke = value; 
+        }
+    }
```

Comment:
You should make all of these into non-nullable auto properties. You do not need a class level property for `Joke`, this can simply be a locally scoped variable to the method it's used in.

```suggestion
    private IComedian Comedian { get; set; }
    private IJokeable Jokeable { get; set; }
```